test/test_linalg.py

```diff
@@ -6014,19 +6043,23 @@ def maybe_transpose(cond, m):
             M = maybe_transpose(t1, torch.randn(10, 25, device=device).to(dtype))
             m1 = maybe_transpose(t2, torch.randn(10, 50, device=device).to(dtype))
             m2 = maybe_transpose(t3, torch.randn(50, 25, device=device).to(dtype))
-            self._test_addmm_addmv(torch.addmm, M, m1, m2, transpose_out=t4)
+            self._test_addmm_addmv(torch.addmm, M, m1, m2, transpose_out=t4, layout=layout)
 
     @dtypes(torch.float, torch.double)
     @dtypesIfCUDA(*([torch.float, torch.double] + torch.testing.get_all_complex_dtypes()))
     @tf32_on_and_off(0.005)
-    def test_addmm_sizes(self, device, dtype):
+    @enableSparseCSR(only_cuda=True,
+                     dtypes_cuda=(*torch.testing.get_all_complex_dtypes(),
```

Comment:
I see there is a `@dtypesIfCUDA` decorator and the `dtypes_cuda` flag. How does the interplay work?

Reply:
`@dtypesIfCUDA` applies only to normal test and `@enableSparseCSR(dtypes_cuda=...)` applies to sparse_csr layout tests. If `dtypes_cuda` is omitted then dtypes specified by `@dtypesIfCUDA, @dtypesIfCPU, @dtypes` are used for both sets of tests.